Controllers/Frontend/StripePayment.php

```diff
@@ -333,34 +482,67 @@ protected function processChargeSucceededEvent(Stripe\Event $event)
     }
 
     /**
-     * First checks the Shopware session for the 'stripePayment->processingSourceId' field and,
-     * if set, makes sure the ID matches the source contained in the event. Then waits for five
-     * seconds to prevent timing issues caused by webhooks arriving earlier than e.g. a redirect
-     * during the payment process. That is, if completing the  payment process involves e.g.
-     * a redirect to the payment provider, the 'source.chargeable' event might arrive at the shop
-     * earlier than the redirect returns. By pausing the webhook handler, we give the redirect a
-     * head start to complete the order creation. After waiting, the database is checked for an
-     * order that used the event's source. If no such order is found, the source is used to
-     * create a charge and the session's order is saved to the database.
+     * Tries to find the order the event belongs to and, if found, update its payment status
+     * to 'review necessary'.
      *
      * @param Stripe\Event $event
      */
-    protected function processSourceChargeableEvent(Stripe\Event $event)
+    protected function processSourceFailedEvent(Stripe\Event $event)
     {
-        // Check whether the webhook event is allowed to create an order
-        $source = $event->data->object;
-        $stripeSession = Util::getStripeSession();
-        if ($source->id !== $stripeSession->processingSourceId) {
+        $order = $this->findOrderForWebhookEvent($event);
+        if (!$order) {
             return;
         }
+        $paymentStatus = $this->get('models')->find(Status::class, Status::PAYMENT_STATE_REVIEW_NECESSARY);
+        $order->setPaymentStatus($paymentStatus);
+        $this->get('models')->flush($order);
+    }
 
+    /**
+     * First wait for five seconds to prevent timing issues caused by webhooks arriving
+     * earlier than e.g. a redirect during the payment process. That is, if completing the payment
+     * process involves e.g. a redirect to the payment provider, the 'source.chargeable' event might
+     * arrive at the shop earlier than the redirect returns. By pausing the webhook handler, we give the redirect a
+     * head start to complete the order creation. After waiting, the database is checked for an
+     * order that used the event's source. If an order is found and its transactionId starts with 'src_pending',
```

Comment:
Based on my comment above:
```suggestion
     * order that used the event's source. If an order is found and its transactionId is 'stripe_source_pending',
```